linux/hid.c

```diff
@@ -582,6 +754,53 @@ struct hid_device_info  HID_API_EXPORT *hid_enumerate(unsigned short vendor_id,
 					 * check for USB and Bluetooth devices above */
 					break;
 			}
+
+			/* Construct <sysfs_path>/device/report_descriptor */
+			char *rpt_path = (char *)calloc(1, strlen(sysfs_path) + 25);
+			sprintf(rpt_path, "%s/device/report_descriptor", sysfs_path);
+
+			/* Usage Page and Usage */
+			int res;
+			struct hidraw_report_descriptor rpt_desc;
+			res = get_hid_report_descriptor(rpt_path, &rpt_desc);
+			if (res >= 0) {
+				unsigned short page = 0, usage = 0;
+				unsigned int pos = 0, usage_count = 0;
+				/*
+				 * Parse the usage and usage page
+				 * out of the report descriptor.
+				 */
+				while (!get_next_hid_usage(rpt_desc.value, rpt_desc.size, &pos, &page, &usage)) {
+					usage_count++;
+
+					/* First usage */
+					if (usage_count == 1)
+					{
+						cur_dev->usage_page = page;
+						cur_dev->usage = usage;
+						continue;
+					}
```

Comment:
have it only once, outside the loop, and loose `usage_count` variable

Reply:
Actually I think I have solution.

Reply:
so do I, in case if you want to compare :)